### DISCUSSION TITLE
Missing 'Next' Sensors

### DISCUSSION THREAD

**Discussion by RexNoctis** [BottlecapDave/HomeAssistant-OctopusEnergy#583] [General]
Hi all,

Since the latest update, I don't appear to have access to the upcoming Agile prices, I can see my past usage with unit rates in the ..._current_accumulative_cost sensor and I also have individual ..._current_rate and ..._next_rate sensors but nothing with the array of future rates.

I do have a number of unavailable sensors but no idea why they are now unavailable? I did try reconfiguring the integration (and adding that I now have a mini).

Any ideas? What further information would be helpful?

Thanks!

Danny.

**Comment by Stephens-ely** [comment]
I have the same issue - the sensor 'sensor.octopus_energy_electricity_19mYYYYYYYYY_XXXXXXXXXXX_current_rate' does not have a attribute filled with the rates.

It's due to changes in V9 which changed  / split sensors and made them events.

**Comment by BottlecapDave** [comment]
As per the docs, the [deprecation notice](https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/discussions/444) and [changelog](https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/blob/develop/CHANGELOG.md#900-2023-11-20) for the last major version, these attributes are no longer available in the rate sensors, but as @Stephens-ely pointed out are now attributes as part of the `event` sensors. These were initially disabled by default, so if you've installed the integration before 9.0.1, you'll need to enable the sensors by following the [FAQ](https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/blob/develop/_docs/faq.md#there-are-entities-that-are-disabled-why-are-they-disabled-and-how-do-i-enable-them)

Depending on your tariff you, as outlined by the changelogs a variety of sensors name's were updated to include the account id. Therefore the old sensors should be deleted, and the new sensors used instead. This is to help support future features.

**Comment by RexNoctis** [comment]
@BottlecapDave - Apologies, I thought I'd been through the changelog, managed to miss this completely! I appreciate your time (and the work that has gone into this integration)!
